### PR TITLE
Update contribution guide with info about forks

### PR DESCRIPTION
**Pull request by erinha**
Adding info to the CONTRIBUTING.md file for how to fork the repository to contribute to the teams-js repo.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,10 +5,12 @@ One of the best ways to contribute is to participate in discussions and discuss
 ## General Contributing Guide
 
 1. Unless it is a trivial change, make sure that there is a corresponding issue for your change first. If there is none, create one.
-2. Create a fork in GitHub
-3. Create a branch off the `master` branch. Name it something that that makes sense, such as `issue-123` or `githubhandle-issue`. This makes it easy for everyone to figure out what the branch is used for. It also makes it easier to isolate your change from incoming changes from the origin.
-4. Commit your changes and push your changes to GitHub
-5. Create a pull request against the origin's `master` branch. If you don't know what a pull request is read [this article](https://help.github.com/articles/using-pull-requests).
+2. Create a [fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks) in GitHub.
+3. Create a branch off the `master` branch in your fork. Name it something that that makes sense, such as `issue-123` or `githubhandle-issue`. This makes it easy for everyone to figure out what the branch is used for. It also makes it easier to isolate your change from incoming changes from the upstream repo.
```
Comment (Jessica-Koch) [inline CONTRIBUTING.md:9]: are we going to have to update this when master dies?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,10 +5,12 @@ One of the best ways to contribute is to participate in discussions and discuss
...
+4. Commit your changes and push your changes to GitHub.
+5. Create a [pull request against the upstream's `master` branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork). If you don't know what a pull request is read [this article](https://help.github.com/articles/using-pull-requests).
+6. A user with write access to the upstream repo will need to trigger the checks that run on pull requests in the teams-js repo.
+7. Once your pull request is approved, a user with write access will need to merge your pull request into the upstream repo. You can then delete your branch.
```
Comment (Jessica-Koch) [inline CONTRIBUTING.md:13]: is upstream repo a thing, I thought upstream only referred to branches, and it was just the main or source repository.

File: CONTRIBUTING.md (hunk 1 above)
Comment (erinha) [reply CONTRIBUTING.md:9]: Nope. I have a separate change for 2.0-preview.

File: CONTRIBUTING.md (hunk 2 above)
Comment (erinha) [reply CONTRIBUTING.md:13]: Good question, it is a thing. From [this](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) page: "You can create a pull request to propose changes you've made to a fork of an upstream repository"

**Review by MarcinJuraszek — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (MarcinJuraszek) [reply CONTRIBUTING.md:13]: I've only ever heard of upstread repos and have never heard of upstream branches :D I need to go look up what that is :)

**Review by Jessica-Koch — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (Jessica-Koch) [reply CONTRIBUTING.md:13]: so I think I am confused by git terms, lol, I found https://stackoverflow.com/questions/57491448/upstream-is-a-repo-or-a-branch and it helped a lot.

Comment (Jessica-Koch) [reply CONTRIBUTING.md:13]: @MarcinJuraszek I think I am mixing up upstream and remotes....I should really study git terminology

**Pull request merged**